sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/exceptions.py

```diff
@@ -5,33 +5,52 @@
 import functools
 from typing import TYPE_CHECKING
 
-from azure.core.exceptions import DecodeError, ResourceExistsError, ResourceNotFoundError
+from azure.core.exceptions import DecodeError, HttpResponseError, ResourceExistsError, ResourceNotFoundError
 from azure.core.pipeline.policies import ContentDecodePolicy
 
 if TYPE_CHECKING:
     # pylint:disable=unused-import,ungrouped-imports
-    from typing import Type
-    from azure.core.exceptions import AzureError
+    from typing import Optional, Type
     from azure.core.pipeline.transport import HttpResponse
 
 
-def get_exception_for_key_vault_error(cls, response):
-    # type: (Type[AzureError], HttpResponse) -> AzureError
+def _get_exception_for_key_vault_error(cls, response):
+    # type: (Type[HttpResponseError], HttpResponse) -> HttpResponseError
+    """Construct cls (HttpResponseError or subclass thereof) with Key Vault's error message."""
+
     try:
         body = ContentDecodePolicy.deserialize_from_http_generics(response)
-        message = "({}) {}".format(body["error"]["code"], body["error"]["message"])
+        message = "({}) {}".format(body["error"]["code"], body["error"]["message"])  # type: Optional[str]
     except (DecodeError, KeyError):
-        # Key Vault error response bodies have the expected shape and should be deserializable.
+        # Key Vault error response bodies should have the expected shape and be deserializable.
         # If we somehow land here, we'll take HttpResponse's default message.
         message = None
 
     return cls(message=message, response=response)
 
 
+# errors map to HttpResponseError...
+_default = functools.partial(_get_exception_for_key_vault_error, HttpResponseError)
+
+# ...unless this mapping specifies another type
 _code_to_core_error = {404: ResourceNotFoundError, 409: ResourceExistsError}
 
+
+class _ErrorMap(dict):
+    """A dict whose 'get' method returns a default value.
+
+    defaultdict would be preferable but defaultdict.get returns None for keys having no value
+    (azure.core.exceptions.map_error calls error_map.get)
```

Comment:
I'm afraid this is way too much specific here, to rely on azure-core doing a "get". I think the contract of azure-core is "dict-like", which then is not the case here. For instance, get has two parameters, so this override is not even valid.

I think I would do something like:
```python
class Foo:
    def __repr__(self):
        return f"I'm foo instance {hex(id(self))}"

class fulldefaultdict(defaultdict):
    def get(self, key, value=None):
        return self[key]

In [17]: t = fulldefaultdict(Foo)

In [18]: t.get(17)
Out[18]: I'm foo instance 0x1f997517da0

In [19]: t[48]
Out[19]: I'm foo instance 0x1f9975178d0

In [20]: t.get(48)
Out[20]: I'm foo instance 0x1f9975178d0

In [21]: t.get(89, "default")
Out[21]: I'm foo instance 0x1f99749b588
```

Reply:
It does not support "in", so it's not enough, let me see if I find somewhere even smarter

Reply:
Is this rely on on azure-core doing a `get`, or accommodating it do ing a `get` :) Why not address this by adding capability to configure `azure.core.exceptions.map_error` for a different default i.e. it could look for a distinguished key like "default" in the error map, that should be a non-breaking addition, and then anyone could utilize this without having to create some custom dict-thing.

Reply:
I agree that's a better long-term solution (#9773). My goal here is a quick fix for the next azure-keyvault release.